substratevm/src/com.oracle.svm.test.jdk11/src/com/oracle/svm/test/jdk11/jfr/utils/JFRFileParser.java

```diff
@@ -0,0 +1,168 @@
+        long metadataPosition = input.readRawLong();
+        assertTrue("Metadata positions is equals to null!", metadataPosition > 0);
+
+        assertTrue("Starting time is null!", input.readRawLong() > 0); // Starting time.
+        input.readRawLong(); // Duration.
+        assertTrue("Chunk start tick is null!", input.readRawLong() > 0); // ChunkStartTick.
+        assertTrue("Tick frequency is null!", input.readRawLong() > 0); // Tick frequency.
+        int shouldUseCompressedInt = input.readRawInt();
+        assertTrue("Compressed int can be either 0 or 1!", shouldUseCompressedInt == 0 || shouldUseCompressedInt == 1); // ChunkWriteTick.
+
+        return new Positions(constantPoolPosition, metadataPosition);
+    }
+
+    private static void parseMetadataHeader(RecordingInput input, long metadataPosition) throws IOException {
+        input.position(metadataPosition); // Seek to starting position of metadata region.
+        assertTrue("Metadata size is null!", input.readInt() > 0); // Size of metadata.
+        assertEquals(JfrChunkWriter.METADATA_TYPE_ID, input.readLong()); // Metadata region ID.
+        assertTrue("Metadata timestamp is null!", input.readLong() > 0); // Timestamp.
+        input.readLong(); // Duration
+        input.readLong(); // Metadata ID.
+    }
+
+    private static void parseMetadata(RecordingInput input, long metadataPosition) throws IOException {
+        parseMetadataHeader(input, metadataPosition);
+        MetadataDescriptor.read(input);
+    }
+
+    private static long parseConstantPoolsHeader(RecordingInput input, long constantPoolPosition) throws IOException {
+        input.position(constantPoolPosition); // Seek to starting position of constant pools.
+        assertTrue("Constant pool size is null!", input.readInt() > 0); // Size of constant pools.
+        assertEquals(JfrChunkWriter.CONSTANT_POOL_TYPE_ID, input.readLong()); // Constant pools region ID.
+        assertTrue("Constant pool timestamp is null!", input.readLong() > 0); // Timestamp.
+        input.readLong(); // Duration.
+        long deltaNext = input.readLong(); // Offset to a next constant pools region.
+        assertTrue(input.readBoolean()); // Flush.
+        return deltaNext;
+    }
+
+    private static void parseConstantPools(RecordingInput input, long constantPoolPosition) throws IOException {
+        long deltaNext = parseConstantPoolsHeader(input, constantPoolPosition);
+        long numberOfCPs = input.readInt();
+        for (int i = 0; i < numberOfCPs; i++) {
+            supportedConstantPools.getOrDefault(input.readLong(), notFoundConstantPoolParser).parse(input);
+        }
+        if (deltaNext != 0) {
+            parseConstantPools(input, constantPoolPosition + deltaNext);
+        }
+    }
+
+    private static void parseEvents(RecordingInput input) {
+        // TODO: We should implement file parsing in the same way for the events (as it is for constant pools)?
```

Comment:
I'd consider re-using the public api `jdk.jfr.consumer` to parse events (e.g. `jdk.jfr.consumer.RecordingFile`)

Reply:
Or at least using it as inspiration for parsing events

Reply:
I'll take a look. :smile: